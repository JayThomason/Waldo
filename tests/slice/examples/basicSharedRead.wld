/**
  * Just checking to ensure that pingCounter appears in the definite
  * read set of something
*/


PingPong

Endpoint Ping;
Endpoint Pong;


Sequences
{
}


/**** Data shared between both ends */
Shared
{
    // both of these are rarely used in this program.
    Ping controls Number pingCounter = 0;
    Pong controls Number pongCounter = 0;
}




/**** Ping endpoint *********/
Ping
{
    Public Function something()
    {
        pingCounter;
    }
}


/**** Pong endpoint *********/
Pong
{
}
