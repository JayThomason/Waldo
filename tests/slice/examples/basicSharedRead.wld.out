

{
    "conditionalGlobalSharedReads": [], 
    "conditionalGlobalSharedWrites": [], 
    "definiteGlobalSharedReads": [
        {
            "argPosition": -1, 
            "id": 0, 
            "mutable": 0, 
            "varName": "pingCounter", 
            "varType": 0
        }
    ], 
    "definiteGlobalSharedWrites": [], 
    "funcName": "something", 
    "readSet": [
        "pingCounter"
    ]
}





